tests/ssl-opt.sh

```diff
@@ -2107,7 +2107,7 @@ run_test    "key size: TLS-ECDHE-ECDSA-WITH-AES-128-CCM-8" \
             -c "Key size is 128"
 
 requires_config_enabled MBEDTLS_X509_CRT_PARSE_C
-requires_config_enabled MBEDTLS_MD_CAN_MD5
+requires_config_enabled PSA_WANT_ALG_MD5
```

Comment:
`PSA_WANT_ALG_MD5` only works in configurations where the TLS layer uses PSA for cryptography. In configurations where the TLS layer uses the legacy crypto API, this test case is omitted. In legacy configurations, the test needs to use `MBEDTLS_MD5_C` instead.

This causes the [errors reported by outcome analysis](https://mbedtls.trustedfirmware.org/job/mbed-tls-pr-head/job/PR-9290-head/1/execution/node/8425/log/?consoleFull): these test cases are passing with PSA drivers, but failing in a legacy-only configuration.

The fix is to use `requires_hash_alg`, as is done for `SHA_256`. That shell function takes care of supporting both crypto interfaces.